src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -224,6 +224,52 @@ func (f *FilesystemShare) Cleanup(ctx context.Context) error {
 	return nil
 }
 
+func readDir(path string) ([]string, error) {
+
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return file.Readdirnames(0)
+}
+
+func (f *FilesystemShare) recursiveCopy(ctx context.Context, src, dst string) (bool, error) {
```

Comment:
Is it possible to use https://pkg.go.dev/path/filepath#WalkDir instead? It should be able to save some lines.

Reply:
done